instrumentation/excon/lib/opentelemetry/instrumentation/excon/middlewares/tracer_middleware.rb

```diff
@@ -71,43 +79,35 @@ def self.around_default_stack
           private
 
           def handle_response(datum)
-            if datum.key?(:otel_span)
-              datum[:otel_span].tap do |span|
-                return span if span.end_timestamp
+            datum.delete(:otel_span)&.tap do |span|
+              return span unless span.recording?
 
-                if datum.key?(:response)
-                  response = datum[:response]
-                  span.set_attribute('http.status_code', response[:status])
-                  span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(response[:status].to_i)
-                end
-
-                span.status = OpenTelemetry::Trace::Status.error("Request has failed: #{datum[:error]}") if datum.key?(:error)
+              if datum.key?(:response)
+                response = datum[:response]
+                span.set_attribute(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE, response[:status])
+                span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(response[:status].to_i)
+              end
 
-                span.finish
-                datum.delete(:otel_span)
+              if datum.key?(:error)
+                span.status = OpenTelemetry::Trace::Status.error("Request has failed: #{datum[:error]}")
```

Comment:
Since we are recording the exception we probably should remove the dynamic portion of the message. I am not sure why the original author would have chosen interpolation instead of recording the error.

Reply:
That makes sense to me. Went ahead and made this message static.